File: testprojects/python_main/pipelines/PERF_PY_PIP/code/com/main1/pythondepmanagement_1/graph/PERF_SG/WindowFunction_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.libs import typed_lit
from com.main1.pythondepmanagement_1.config.ConfigStore import *
from com.main1.pythondepmanagement_1.udfs.UDFs import *

def WindowFunction_1(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0\
        .withColumn(
          "cmls_acct_fundg_srce_cd_drvd",
          row_number()\
            .over(Window.partitionBy(col("cmls_acct_cobrnd_bus_id_drvd")).orderBy(col("cmls_acct_cobrnd_bus_id_drvd").desc()))
        )\
        .withColumn("cmls_acct_fundg_srce_cd_enr", row_number()\
        .over(Window.partitionBy(col("cmls_acct_cobrnd_bus_id_drvd")).orderBy(col("cmls_acct_cobrnd_bus_id_drvd").desc())))

File: testprojects/scala_main/pipelines/DONOT_OPEN_SCALA/code/src/main/scala/org/main/scla_dep_mgmt/graph/WindowFunction_1.scala
package org.main.scla_dep_mgmt.graph

import io.prophecy.libs._
import org.main.scla_dep_mgmt.udfs.UDFs._
import org.main.scla_dep_mgmt.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object WindowFunction_1 {

  def apply(context: Context, in: DataFrame): DataFrame = {
    import org.apache.spark.sql.expressions.{Window, WindowSpec}
    in.withColumn(
        "c_date-for today",
        row_number().over(
          Window
            .partitionBy(col("`c- short`"),
                         col("`c  - int`"),
                         col("`- c long`")
            )
            .orderBy(col("`c_float-__  `").asc,
                     col("`c -  boolean _  `").asc,
                     col("c_double").asc,
                     col("`c-string`").asc
            )
        )
      )
      .withColumn(
        "c_timestamp  __ for--today",
        row_number().over(
          Window
            .partitionBy(col("`c- short`"),
                         col("`c  - int`"),
                         col("`- c long`")
            )
            .orderBy(col("`c_float-__  `").asc,
                     col("`c -  boolean _  `").asc,
                     col("c_double").asc,
                     col("`c-string`").asc
            )
        )
      )
  }

}
